doc/source/core_array_funcs.md

```diff
@@ -0,0 +1,128 @@
+the output will be a {class}`~geoutils.Raster`. If there is a second input, it can be a {class}`~geoutils.Raster` or {class}`~numpy.ndarray` with
+matching georeferencing or shape, respectively.
+
+These functions inherently support the casting of different {attr}`~geoutils.Raster.dtypes` and values masked by {attr}`~geoutils.Raster.nodata` in the
+{class}`~numpy.ma.MaskedArray`.
+
+Below, we re-use the same example created in {ref}`core-py-ops`.
+
+```{code-cell} ipython3
+:tags: [hide-input, hide-output]
+
+import geoutils as gu
+import rasterio as rio
+import pyproj
+import numpy as np
+
+# Create a random 3 x 3 masked array
+np.random.seed(42)
+arr = np.random.randint(0, 255, size=(3, 3), dtype="uint8")
+mask = np.random.randint(0, 2, size=(3, 3), dtype="bool")
+ma = np.ma.masked_array(data=arr, mask=mask)
+
+# Create an example Raster
+raster = gu.Raster.from_array(
+        data = ma,
+        transform = rio.transform.from_bounds(0, 0, 1, 1, 3, 3),
+        crs = pyproj.CRS.from_epsg(4326),
+        nodata = 255
+    )
+raster
+```
+
+```{code-cell} ipython3
+# Universal function with a single input and output
+np.sin(raster)
+```
+
+```{code-cell} ipython3
+# Universal function with a two inputs and single output
+np.add(arr, raster)
+```
+
+```{code-cell} ipython3
+# Universal function with a single input and two outputs
+np.modf(raster)
+```
+
+Similar to with Python operators, NumPy's [logical comparison functions](https://numpy.org/doc/stable/reference/ufuncs.html#comparison-functions) cast
+{class}`Rasters<geoutils.Raster>` to a {class}`~geoutils.Mask`.
+
+```{code-cell} ipython3
+# Universal function with a single input and two outputs
+np.greater(raster, raster + np.random.normal(size=np.shape(arr)))
+```
+
+## Array functions
+
+The second and last category of NumPy array functions supported by {class}`Rasters<geoutils.Raster>` through the array interface is that of array functions,
+which are all other non-universal functions that can be applied to an array. Those function always modify the dimensionality of the output, such as
+{func}`~numpy.mean`, {func}`~numpy.count_nonzero` or {func}`~numpy.nanmax`. Consequently, the output is the same as it would be with {class}`ndarrays<numpy.ndarray>`.
+
+
+```{code-cell} ipython3
+# Traditional mathematical function
+np.max(raster)
+```
+
+```{code-cell} ipython3
+# Expliciting an axis for reduction
+np.count_nonzero(raster, axis=2)
+```
+
+Not all array functions are supported, however. GeoUtils supports nearly all [mathematical functions](https://numpy.org/doc/stable/reference/routines.math.html),
+[masked-array functions](https://numpy.org/doc/stable/reference/routines.ma.html) and [logical functions](https://numpy.org/doc/stable/reference/routines.logic.html).
+A full list of supported array function is available in {attr}`geoutils.raster.raster.handled_array_funcs`.
```

Comment:
I'm surprised by the double "raster" in the submodule name.

Reply:
Fixed